README.md

```diff
@@ -1,45 +1,97 @@
-
-## Documents
-
-- [**Roadmap**](/ROADMAP.md)
-- **Specifications:**
-  - ![](https://img.shields.io/badge/status-draft-yellow.svg?style=flat-square) [`IPLD`](/IPLD.md) - spec about the data model, pointers and link formats
-  - ![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square) `IPLD Selectors` - spec about simple language to select multiple unknown nodes in a graph
-  - ![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square) `IPLD Transformations` - spec about the language to trasform an IPLD graph into another
-  - ![](https://img.shields.io/badge/status-reliable-green.svg?style=flat-square) [`CID (Content IDentifier)`](https://github.com/ipld/cid)
-  - ![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square) [`IPLD Formats`](https://github.com/ipld/interface-ipld-format) - interface definition for adding support to different formats
-  - ![](https://img.shields.io/badge/status-draft-yellow.svg?style=flat-square) [`CAR`](/CAR.md) - Content Addressable Archives
+IPLD is not a single specification, it is a set of specifications.
+
+```
+                        The IPLD Stack
+                                +-----------------------------+
+                +-------------+ |                             |
+                |             | | End-User Application Stacks |
+                | MFS in IPFS | |                             |
+                |             | +-----------------------------+
+                +-------------+ |                             |
+                |             | | Structured Data w/ indexes  |
+                |  unixfs v2  | |     VR, Geo, SQL, etc.      |     +----------+
+                |             | |                             |     |          |
+                +-------------+ +-----------------------------+     |  MFS in  |
+                |             | |                             |     |   IPFS   |
+                |    HAMT     | |   Sorted Index (sharded)    |     |          |
+                |             | |                             |     +----------+
+                +-------------+-+-----------------------------+     |          |
+                |                                             |     |  unixfs  |
+                |          Complex Data Structures            |     |    v1    |
+                |                                             |     |          |
++------------------------------------------------------------------------------+
+|               |                                             |     |          |
+|               |        dag-json         dag-cbor            |     |          |
+|               |                                             |     |          |
+|    Codecs     +---------------------------------------------+ git |  dag-pb  |
```

Comment:
@mikeal I may be misunderstanding a lot here so sorry in advance but `git` seems to be out of place here? If you were to classify `ipld-eth`, `ipld-btc`, and `ipld-zcash` would they also go here or would they go above `Complex Data Structures`?

Reply:
git, eth, btc and zcash are all in the same level as `git` is. these are all pre-existing formats that don't support all the simple types necessary to implement **our** higher order data-structures like `hamt`. That is not to say they can't implement some sort of hamt, they may already, but they can't implement the generic hamt data-structure we are going to spec out. 

perhaps we should label that section `ipld-git` and put a few more in the same box like `ipld-btc`.

Reply:
> perhaps we should label that section ipld-git and put a few more in the same box like ipld-btc.

This would make it clearer :+1:.

To clarify my understanding a little further, if I were to implement an 'ipld-format' similar to `ipld-git` now, where should it go? Should I be looking to implement something in the way that `ipld-git` or `ipld-btc` does?

Reply:
@lanzafame it depends :)

But just to make this a little clearer. The general decoders, like the ones we use in IPFS, don't really differ between formats with the full data model and those without. The API for that layer will remain simple so that it's easy to implement future codecs like `ipld-git` for formats that already exist.

However, if you want to build a codec that supports the full data model so that you can use all the complex data-structures we're building on top of that (like `hamt`) then you'll want to support the full data model in your encoder. This just means being able to encode and decode a number of standard types.

Reply:
I've been calling these sorts of things "IPLD-Adjunct" formats: data that we can read and address as part of the merkleforest, but it requires special code to do so, rather than being an "IPLD-native" system which works only with our totally reusable cbor/json/etc systems.

Reply:
New visualization is better :)